flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/KubernetesOperatorConfigOptions.java

```diff
@@ -410,6 +430,14 @@ public static String operatorConfigKey(String key) {
                     .withDescription(
                             "Max number of reconcile loops triggered within the rate limiter refresh period for each resource. Setting the limit <= 0 disables the limiter.");
 
+    @Documentation.Section(SECTION_DYNAMIC)
+    public static final ConfigOption<Boolean> OPERATOR_JOB_UPGRADE_SAVEPOINT_DISPOSE_ON_DELETE =
+            operatorConfig("job.upgrade.savepoint.dispose-on-delete")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Savepoint data created by upgrade savepoints as FlinkStateSnapshot resources will be disposed of automatically when the Kubernetes resource gets deleted.");
+
```

Comment:
I think we should have a single dispose-on-delete config that generally applies to snapshot resources.

Reply:
Good idea, I have named it `savepoint.dispose-on-delete`.